src/resolve/resolved-dns-query.c

```diff
@@ -524,6 +573,11 @@ int dns_query_new(
 
         assert(m);
 
+        /* Check for records that is refused and refuse query for the records if matched in configuration */
+        if (dns_question_has_record_types(question_utf8, m->refuse_record_types) ||
```

Comment:
Please directly call `manager_validate_and_mangle_question()` and propagate the error code:
```c
r = validate_and_mangle_question(question_utf8, m->refuse_record_types);
if (r < 0)
        return r;

r = validate_and_mangle_question(question_idna, m->refuse_record_types);
if (r < 0)
        return r;
```